ui/src/components/TasksList.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { generatePath, useNavigate } from 'react-router-dom';
+import PickerDifficultyLevel from '@components/PickerDifficultyLevel';
+import TaskCard from '@components/TaskCard';
+import paths from '@routes/paths';
+
+interface Task {
+  id: number;
+  name: string;
+  description: string;
+  image: string;
+  difficulty: number;
+}
+
+const filterTasksByDifficulty = (tasks: Task[], difficultyLevel: number) => {
+  return tasks.filter((task) => task.difficulty === difficultyLevel);
+};
+
+const TasksList = ({ tasks }: { tasks: Task[] }) => {
+  const [difficultyLevel, setDifficultyLevel] = useState(1);
+  const [tasksFilter, setTaskFilter] = useState(filterTasksByDifficulty(tasks, difficultyLevel));
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setTaskFilter(tasks.filter((task) => task.difficulty === difficultyLevel));
+  }, [difficultyLevel, tasks]);
```

Comment:
to też niepotrzebne, możemy od razu w linii 38 robić task.filter().map() i od razu to renderować. Ten useEffect wprowadza tutaj niepotrzebny chaos